.github/workflows/upload-testing-image-to-ecr.yaml

```
@@ -0,0 +1,47 @@
+name: Upload testing image to ECR
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - tests/**
+  workflow_dispatch:
+    branches:
+      - main
+
+env:
+  AWS_REGION: eu-west-2
+
+jobs:
+  upload-testing-artifact:
+    name: Upload testing image to ECR
+    permissions:
+      id-token: write
+      contents: read
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Assume AWS role
+        uses: aws-actions/configure-aws-credentials@v1-node16
```

```suggestion
        uses: aws-actions/configure-aws-credentials@v2
```

The v1-node16 action is now deprecated. Should probably update this elsewhere too!